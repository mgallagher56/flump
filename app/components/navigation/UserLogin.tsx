import type { FC } from 'react';

import { useLoaderData } from '@remix-run/react';
import { useTranslation } from 'react-i18next';
import type { loader } from '~/root';

import FLPButtonGroup from '../core/buttons/FLPButtonGroup';
import FLPLinkButton from '../core/buttons/FLPLinkButton';
import SignOut from '../users/SignOut';

const UserLogin: FC = () => {
  const { t } = useTranslation();
  const { user } = useLoaderData<typeof loader>();

  return (
    <FLPButtonGroup>
      {user ? (
        <SignOut />
      ) : (
        <>
          <FLPLinkButton to="/signup" variant="solid">
            {t('signUp')}
          </FLPLinkButton>
          <FLPLinkButton to="/login" variant="outline">
            {t('logIn')}
          </FLPLinkButton>
        </>
      )}
    </FLPButtonGroup>
  );
};

export default UserLogin;
