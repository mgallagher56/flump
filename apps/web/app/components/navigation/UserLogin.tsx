import type { FC } from 'react';

import { useTranslation } from 'react-i18next';
import { useLoaderData } from 'react-router';
import FLPButtonGroup from '~/components/core/buttons/FLPButtonGroup';
import FLPLinkButton from '~/components/core/buttons/FLPLinkButton';
import SignOut from '~/components/users/SignOut';
import type { loader } from '~/root';

const UserLogin: FC = () => {
  const { t } = useTranslation();

  const { user } = useLoaderData<typeof loader>();

  return (
    <FLPButtonGroup>
      {user ? (
        <SignOut />
      ) : (
        <>
          <FLPLinkButton to="/signup" variant="solid">
            {t('signUp')}
          </FLPLinkButton>
          <FLPLinkButton to="/login" variant="outline">
            {t('logIn')}
          </FLPLinkButton>
        </>
      )}
    </FLPButtonGroup>
  );
};

export default UserLogin;
