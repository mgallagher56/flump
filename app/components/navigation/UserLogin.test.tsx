import { render } from '@testing-library/react';
import { vi } from 'vitest';
import UserLogin from '~/components/navigation/UserLogin';

import mockUser from '__mocks__/user';

const mocks = vi.hoisted(() => ({
  mockUseLoaderData: vi.fn()
}));

vi.mock('@remix-run/react', async () => {
  const actual: Record<string, unknown> = await vi.importActual('@remix-run/react');
  return {
    ...actual,
    useLoaderData: mocks.mockUseLoaderData
  };
});

vi.mock('app/components/users/SignOut', () => ({
  default: () => <div>logOut</div>
}));

vi.mock('app/components/core/buttons/FLPLinkButton', () => ({
  default: (props) => <div>{props.children}</div>
}));

describe('<UserLogin />', () => {
  test('renders as expected when user not logged in', () => {
    mocks.mockUseLoaderData.mockReturnValue({ user: undefined });
    const { baseElement } = render(<UserLogin />);
    expect(baseElement).toMatchSnapshot();
  });
  test('renders signed out when logged in', () => {
    mocks.mockUseLoaderData.mockReturnValue({ user: mockUser });
    const { getByText } = render(<UserLogin />);
    expect(getByText('logOut')).toBeDefined();
  });
});
