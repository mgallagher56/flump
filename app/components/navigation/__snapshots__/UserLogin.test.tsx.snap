// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserLogin /> > renders as expected 1`] = `
<body>
  <div>
    <div
      class="chakra-button__group css-1ind3v2"
      data-orientation="horizontal"
      role="group"
    >
      <button
        class="chakra-button css-1fysbuj"
        id="pressable:pressableBaseButton"
        type="button"
      >
        signUp
      </button>
      <button
        class="chakra-button css-1fysbuj"
        id="pressable:pressableBaseButton"
        type="button"
      >
        logIn
      </button>
    </div>
  </div>
</body>
`;
